src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -162,6 +163,22 @@ protected ActionResult execute() throws EntityDoesNotExistException {
                 errors.add(String.format(Const.StatusMessages.FEEDBACK_RESPONSE_INVALID_RECIPIENT, questionIndx));
             }
 
+            // If there are duplicate values in recipient list (except empty strings),
+            // trigger this error.
+            for (String recipient : responsesRecipients) {
+                // If response for a recipient is left out, the recipient is then empty string,
+                // due to partial submission feature, there can be multiple recipient with no resposnes,
+                // in that case, there can be multiple empty strings which should not be counted.
+                if ("".equals(recipient)) {
+                    continue;
+                }
+                int frequency = Collections.frequency(responsesRecipients, recipient);
```

Comment:
Also, if the purpose is solely to detect duplicates, another way is to dumping the whole list into a `Set` and compare sizes. What do you think? :D

Reply:
Did that in 4f1771e3dcf9b5ca2dd7f1f6f6c91f2da57a54ef, but actually a problem arises in the following case:

If a student partially answers a question (imagine in questions where a user will have to choose 3 recipients out of a few, and the user chooses 1 and leaves the other two fields blank), then the fields unanswered are denoted by an empty string and that gets added to `responsesRecipients` list. Now, when I use set for size comparison, it causes problem. 

if the user has left two fields blank, and answered for one field, in that case:
`responsesRecipients = [ , , john]`
whereas `responsesRecipientsSet = [ , john]`  (As set will eliminate duplicate empty strings as well)
which will not let the answer get submitted, which contradicts our partial answers can be saved feature. 

So I thought of using this logic as, if I had to use set, I will have to use a very complicated logic, something like, count the number of empty string from `responsesRecipients`, and then
`responsesRecipients - number of empty string` should be equal to `responsesRecipientsSet - 1`.

Reply:
Ah I see 👍